Lib/test/test_strptime.py

```diff
@@ -292,6 +292,10 @@ def test_hour(self):
         self.assertTrue(strp_output[3] == self.time_tuple[3],
                         "testing of '%%I %%p' directive failed; '%s' -> %s != %s" %
                          (strf_output, strp_output[3], self.time_tuple[3]))
+        am_date = (1999, 3, 17, 10, 44, 55, 2, 76, 0)
+        pm_date = (1999, 3, 17, 22, 44, 55, 2, 76, 0)
+        self.assertEqual(_strptime._strptime_time(time.strftime("%I %p", am_date), "%I %p")[3], 10)
+        self.assertEqual(_strptime._strptime_time(time.strftime("%I %p", pm_date), "%I %p")[3], 22)
```

Comment:
I'm not crazy about testing implementation details like `_strptime`, though I realize that that's what this `test_strptime` file does.

Personally, I think it might be worth migrating more of this stuff into `datetimetester.py` and just testing with `datetime.strftime`/`datetime.strptime` instead.

Reply:
Also, the special casing is happening at 0:XX and 12:XX, so you may want to add tests for that.

Reply:
Because of the interaction of `_strptime.py` and `_datetimemodule.c` in the creation of the `datetime` modules, I think it's worthwhile to have separate tests for each of them. We may very well want to test this functionality in `datetimetester.py` as well, but I think we would want to deliberately duplicate the tests in that case (unless we go so far as to remove `test_strptime.py` altogether).

And I think my motivation for picking 10 was from turning 22 into AM, and 22 was likely chosen because it was used in `__calc_am_pm()`. But you raise a valid point about the noon and midnight special cases; I guess I may as well test all 24 hours.